_studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp

```diff
@@ -1675,7 +1675,9 @@ namespace MPEG2EncoderHW
             {
                 return MFX_ERR_UNDEFINED_BEHAVIOR;
             }
-            bWarning = bWarning || (!m_InputSurfaces.CheckInputFrame(&surface->Info));
+            bWarning = bWarning ||
```

Comment:
And regarding previous discussion about codec alignment - it should be an error instead of warning.

Reply:
And looks like best way is change handling of bWarning at the end of the method.

Reply:
@gkrivor I updated the patch but didn't change the handling of bWarning because it is related to  many parameter checks.  Could you please review the patch again?